Tests/Shared.Specs/Xml/XDocumentAssertionSpecs.cs

```diff
@@ -56,6 +56,21 @@ public void When_asserting_a_null_xml_document_is_equal_to_another_xml_document_
             act.Should().Throw<XunitException>();
         }
 
+        [Fact]
+        public void When_a_null_xml_document_is_equal_to_a_null_xml_document_it_should_succeed()
+        {
+            // Arrange
+            XDocument document = null;
+            XDocument otherXDocument = null;
+
+            // Act
+            Action act = () =>
+                document.Should().Be(otherXDocument);
```

Comment:
Every time I see this I wonder whether this should succeed or throw an `ArgumentNullException`.

Reply:
That would be inconsistent with `==` and `Equals(object, object)`. 
Equality in C# is indeed hard.
As often Eric Lippert has blogged about it. https://ericlippert.com/2014/01/15/inconsistent-equality/